master/docs/manual/installation/worker.rst

```diff
@@ -176,13 +176,35 @@ Other Worker Configuration
     The default value is what Python's :func:`sys.getfilesystemencoding()` returns, which on Windows is 'mbcs', on Mac OSX is 'utf-8', and on Unix depends on your locale settings.
 
     If you need a different encoding, this can be changed in your worker's :file:`buildbot.tac` file by adding a ``unicode_encoding`` argument to the Worker constructor.
-
+    
 .. code-block:: python
 
     s = Worker(buildmaster_host, port, workername, passwd, basedir,
                keepalive, usepty, umask=umask, maxdelay=maxdelay,
                unicode_encoding='utf-8', allow_shutdown='signal')
 
+.. _Worker-TLS-Config:
+
+Worker TLS Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+``conndescr``
+    For TLS connections to the master the ``conndescr``-argument must be used to the Worker-initializer. ``buildmaster_host`` and ``port`` must then be ``None``.
+
+    ``conndescr`` should be a string with colon seperated arguments passed to Twisted endpoints ``clientFromString``. An example is ``"TLS:buildbot-master.com:9989"``.
+
+    See more about how to formulate the connection description in `Twisted documentation https://twistedmatrix.com/documents/current/core/howto/endpoints.html`.
+
+    Make sure the worker trusts the masters certificate. If you have an self-signed certificate the trustRoot parameter can be used: ``'TLS:buildbot-master.com:9989:trustRoots=/dir-with-ca-certs'``. It must point to a directory with PEM-encoded certificates in files with file ending .pem.
+
+    ``clientFromString`` supports using certificate and privateKey, but the server does not verify.
```

Comment:
that is client authentication? please clarify

Reply:
Twisted endpoints clientFromString supports setting client site certificates. Which can be really handy for authentication inte an private network via firewall like Big-IP.

The server side of twisted implementation does not offer to verify who issued the client certificate. Which is lacking for completeness.

Regarding the trustRoots is there no error message if the client does not verify the server. It just tries to reconnect again after 1-2s. Maybe is something that can be fixed.

Reply:
I've removed the weird bit about using client certificates. It is just confusing.